unstructured/utils.py

```diff
@@ -712,3 +714,64 @@ def catch_overlapping_and_nested_bboxes(
                 document_with_overlapping_flag = True
 
     return document_with_overlapping_flag, overlapping_cases
+
+
+def get_homedir() -> str:
+    if sys.platform == "win32" and "APPDATA" in os.environ:
+        homedir = os.environ["APPDATA"]
+    else:
+        homedir = os.path.expanduser("~/")
+        if homedir == "~/":
+            raise ValueError("Could not find a default directory for TreeSitter")
+    return homedir
+
+
+def update_tree_sitter(languages: List[str], cache_dir: Optional[str] = None):
```

Comment:
@Plemeur on this bit, I'm inclined to just use the [tree-sitter-languages Python package](https://github.com/grantjenks/py-tree-sitter-languages) as a dependency, rather than enter the business of compiling TreeSitter parsers. Did you consider that option and reject it for some reason?

Reply:
I did not even check if something like that existed since there was the compiling option, it does seems like an easier solution for distribution.